apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -306,67 +355,91 @@ export class PackageJsonUpdater {
 
     await installManager.ensureLocalPackageManager();
     let selectedVersion: string | undefined;
+    const rushConfigProjects: RushConfigurationProject[] = this._rushConfiguration.projects;
 
     if (initialSpec && initialSpec !== 'latest') {
       console.log(colors.gray('Finding versions that satisfy the selector: ') + initialSpec);
       console.log();
-      console.log(`Querying registry for all versions of "${packageName}"...`);
 
-      let commandArgs: string[];
-      if (this._rushConfiguration.packageManager === 'yarn') {
-        commandArgs = ['info', packageName, 'versions', '--json'];
+    // determine if the package is a project in the local repository and if the version exists
+      const version: string | undefined = this.isLocalProject(initialSpec, packageName, projects, rushConfigProjects);
+      if(version !== undefined) {
+        selectedVersion = version
       } else {
-        commandArgs = ['view', packageName, 'versions', '--json'];
+        throw new Error(`The dependency being added ("${packageName}") is a project in the local Rush repository,
```

Comment:
This will throw if the dependency being added isn't a local dependency. In that case, we want to just query the feed.